src/ast/passes/semantic_analyser.cpp

```diff
@@ -462,10 +472,15 @@ void SemanticAnalyser::visit(Call &call)
 
   if (call.vargs) {
     for (size_t i = 0; i < call.vargs->size(); ++i) {
-      auto &expr = (*call.vargs)[i];
+      auto &expr = *(*call.vargs)[i];
       func_arg_idx_ = i;
 
-      expr->accept(*this);
+      if (expr.is_map && skip_key_validation(call)) {
+        Map &map = static_cast<Map &>(expr);
+        map.skip_key_validation = true;
```

Comment:
I guess `call.vargs` is always true for map expressions?

Reply:
Yeah it will be, because we're looking for functions which are taking a map as an argument here (e.g. `clear(@map)`) and `vargs` holds the functions arguments. There's the check for `call.vargs` on line 473 so it should be safe if that's what you were thinking about